storage/mocks/Factory.go

```diff
@@ -111,5 +111,9 @@ func (_m *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger
 
 	return r0
 }
+func (_m *Factory) Purge() error {
+	return nil
+}
 
 var _ storage.Factory = (*Factory)(nil)
+var _ storage.Purger = (*Factory)(nil)
```

Comment:
we don't want to do that. Purger is an optional interface, by force-adding it to the mock you cannot test cases when factory does not implement it. Instead, in your test you can do this:
```
type PurgerFactory struct {
	factoryMocks.Factory
	err error
}

var _ storage.Purger = (*PurgerFactory)(nil)

func (f *PurgerFactory) Purge() error {
	return f.err
}
```